### PR TITLE
documentation: (toy) add initial lowering to riscv

### PR DESCRIPTION
**Pull request by superlopuh**
Adds the structure of the riscv lowering and the first pass. I added some future changes as TODOs so that the remaining work is clear.

### CODE REVIEW

File: docs/Toy/toy/rewrites/setup_riscv_pass.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+from xdsl.passes import ModulePass
+from xdsl.pattern_rewriter import (
+    PatternRewriter,
+    PatternRewriteWalker,
+    RewritePattern,
+    op_type_rewrite_pattern,
+)
+
+
+class AddSections(RewritePattern):
```
Comment (superlopuh) [inline docs/Toy/toy/rewrites/setup_riscv_pass.py:13]: With @AntonLydike's riscemu malloc and friends, the .bss section hack is not necessary, similarly we can probably get away with dropping the data section. I'd like to first merge the end-to-end flow as it's working today, and make it better later, if that's ok with everyone.

File: docs/Toy/toy/compiler.py [hunk 2]
```diff
@@ -83,4 +96,22 @@ def transform(
...
+    SetupRiscvPass().apply(ctx, module_op)
+    # LowerFuncToRiscvFunc().apply(ctx, module_op)
+    # LowerToyAccelerator().apply(ctx, module_op)
+    # LowerMemrefToRiscv().apply(ctx, module_op)
+    # LowerPrintfRiscvPass().apply(ctx, module_op)
+    # LowerArithRiscvPass().apply(ctx, module_op)
+    DeadCodeElimination().apply(ctx, module_op)
+    # ReconcileUnrealizedCastsPass().apply(ctx, module_op)
+
+    DeadCodeElimination().apply(ctx, module_op)
```
Comment (AntonLydike) [inline docs/Toy/toy/compiler.py:110]: twice?

**Review by AntonLydike — APPROVED** [review]
LGTM!

File: docs/Toy/toy/compiler.py (hunk 2 above)
Comment (superlopuh) [reply docs/Toy/toy/compiler.py:110]: Ah yes I guess not really necessary right now, I'll remove it in the next PR

**Pull request merged**